src/components/Dropdown/Dropdown.jsx

```diff
@@ -240,14 +243,16 @@ const Dropdown = ({
     })
   };
 
+  const menuRef = useRef();
   const closeMenuOnScroll = useCallback(
     event => {
       const scrolledElement = event.target;
-      const dropdownContainer = document.getElementById(id);
-      if (dropdownContainer?.contains(scrolledElement)) return false;
+      if (scrolledElement?.parentElement?.classList.contains(MENU_WRAPPER_CLASS_NAME)) {
```

Comment:
for some reason dropdownContainer = document.getElementById(id); return the same component but without the menu (the state of the component before opening the menu) which caused if (dropdownContainer?.contains(scrolledElement) to be false.

I can't pass there ref or id and prefer not changing the nodes structure to avoid potential css overrides issues so I choose to identify the menu by it's classes even it is not the best practice for this kind of issues.